### ISSUE TITLE
drive compilation returns different go versions

### ISSUE THREAD

**Issue by jean-christophe-manciot** [odeke-em/drive#786]
```
drive version
drive version: 0.3.8.1
Commit Hash: 'aa79e324a22a2073b68c61783f4015426b7dbe67'
Go Version: go1.7.1
OS: linux/amd64
BuildTime: 2016-10-19 06:46:40.943196228 +0200 CEST
```
although:
```
go version
go version go1.7.3 linux/amd64
```

**Comment by odeke-em** [comment]
Hello there @jean-christophe-manciot, so the version shown in that commit is the version of Go that built drive after you ran `drive-gen` generated directly from runtime.Version() https://github.com/odeke-em/go-utils/blob/master/pkger/src/pkger.go#L62.

Did you build drive on your computer then later upgrade or did you get it prepackaged?

**Comment by jean-christophe-manciot** [comment]
OK, yes to part 1.

**Comment by jean-christophe-manciot** [comment]
"Built with go version..." would be a more accurate comment.

**Comment by odeke-em** [comment]
I totally agree. Let's reopen this issue. If you are interested, please feel free if you'd like to make
a PR to change the  wording here https://github.com/odeke-em/drive/blob/master/drive-gen/build.go#L61, if not, please let me know and anyone can add it in.

**Comment by odeke-em** [comment]
Rather here https://github.com/odeke-em/go-utils/blob/d915395a7a46a9fe73d93f4daeff5953eeac5ef2/pkger/src/pkger.go#L23

**Comment by jean-christophe-manciot** [comment]
I have 0 experience in Go programming, and I have no intention to gain any.

**Comment by jean-christophe-manciot** [comment]
However, I'm currently writing some bash scripts to automate drive operations. They are not quite ready but I'll let you know when they are.

**Comment by jean-christophe-manciot** [comment]
More information regarding incorrect drive version when compiled with go 1.7.3:
My go is not in $GOPATH/bin:
```
ll $GOPATH/bin/go*
-rwxr-x--- 1 actionmystique actionmystique 2460475 Jul 23 17:01 /home/actionmystique/Program-Files/Ubuntu/GO-Packages/bin/go-bindata*
-rwxr-x--- 1 actionmystique actionmystique 1997121 Jul 23 17:01 /home/actionmystique/Program-Files/Ubuntu/GO-Packages/bin/go-bindata-assetfs*
-rwxr-x--- 1 actionmystique actionmystique 7862620 Jul 31 08:37 /home/actionmystique/Program-Files/Ubuntu/GO-Packages/bin/godeps*
-rwxr-x--- 1 actionmystique actionmystique 6198096 Apr  1  2016 /home/actionmystique/Program-Files/Ubuntu/GO-Packages/bin/godex*
-rwxr-x--- 1 actionmystique actionmystique 5325304 Apr  1  2016 /home/actionmystique/Program-Files/Ubuntu/GO-Packages/bin/goimports*
-rwxr-x--- 1 actionmystique actionmystique 7506320 Apr  1  2016 /home/actionmystique/Program-Files/Ubuntu/GO-Packages/bin/gomvpkg*
-rwxr-x--- 1 actionmystique actionmystique 6817368 Apr  1  2016 /home/actionmystique/Program-Files/Ubuntu/GO-Packages/bin/gorename*
-rwxr-x--- 1 actionmystique actionmystique 6986984 Apr  1  2016 /home/actionmystique/Program-Files/Ubuntu/GO-Packages/bin/gotype*
-rwxr-x--- 1 actionmystique actionmystique 3090544 Jul 23 17:01 /home/actionmystique/Program-Files/Ubuntu/GO-Packages/bin/gox*
```
but rather in $GOROOT/bin (=$GOBIN) as pointed out by the official [go documentation](https://golang.org/doc/install/source#environment):
```
ll $GOROOT/bin/go*
-rwxr-xr-x 1 root root 9953852 Nov 11 13:27 /usr/lib/go/bin/go*
-rwxr-xr-x 1 root root 3036195 Nov 11 13:27 /usr/lib/go/bin/gofmt*

which go
/usr/lib/go/bin/go
```
Also, when compiled with drive-gen without keyword drive-google, the commit hash is now correct, but the go version remains incorrect:
```
drive version
drive version: 0.3.8.1
Commit Hash: '9ce18db69196f53da028fab0ba9621dc33a23656'
Go Version: go1.7.1
OS: linux/amd64
BuildTime: 2016-11-22 09:45:09.58465195 +0100 CET
```

**Comment by odeke-em** [comment]
@jean-christophe-manciot, the symptoms of  https://github.com/odeke-em/drive/issues/794 and even this report seem to me like something isn't right
in your GOPATH setup because somehow you compile with one Go version,
but the binary has another version more than twice.

A simple test for this hypothesis is to compile and run this program in two batches
https://play.golang.org/p/a7AJrBxteb
```go
package main

import (
	"fmt"
	"runtime"
)

func main() {
	fmt.Printf("Go Version: %s\nOS: %s\nArch: %s\n", runtime.Version(), runtime.GOOS, runtime.GOARCH)
}
```

### 1 plain go compile
```shell
$ mkdir 786-repro
$ # Then paste the above program into 786-repro/main.go
$ go build -o 786-repro -o ./786-repro
$ ./786-repro
```

### 2 go get compile
```shell
$ outdir=$GOPATH/src/github.com/drive-repro
$ cd $outdir
$ # The paste the above program in main.go
$ go get .
$ 786-repro
```

**Comment by jean-christophe-manciot** [comment]
This issue has been fixed with 0.3.9:
```
# drive version
drive version: 0.3.9
Commit Hash: 'a9f53bc4cc9d9d2eada836602ad4b4a7902424a6'
Go Version: go1.7.4
OS: linux/amd64
BuildTime: 2017-01-07 09:50:26.013568466 +0100 CET

# go version
go version go1.7.4 linux/amd64
```
